ClientPlugin/Config.cs

```diff
@@ -0,0 +1,57 @@
+using ClientPlugin.Settings.Elements;
+using Sandbox.Graphics.GUI;
+using System;
+using System.Text;
+using System.Xml.Serialization;
+using VRage.Input;
+using VRageMath;
+
+
+namespace ClientPlugin.Settings
+{
+    public enum DropDownEnum
+    {
+        Alpha,
+        Beta,
+        Gamma,
+        Delta,
+        Epsilon
+    }
+
+    public class Config
+    {
+        public static readonly Config Default = new Config();
+        public static readonly Config Current = Storage.Load();
+
+        // Build your UI
+        public readonly string Title = "Config";
+
+        [Keybind("Test Keybind Tooltip")]
+        public MyKeys TestKeybind { get; set; } = MyKeys.None;
+
+        [Textbox("Test Text Input Tooltip")]
+        public string TextInput { get; set; } = "Default Text";
+
+        [Dropdown(description: "Test Dropdown Tooltip")]
+        public DropDownEnum DropdownTest { get; set; } = DropDownEnum.Alpha;
+
+        [Slider(-5f, 4.5f, 0.5f, SliderAttribute.SliderType.Float, description: "Test Float Slider Tooltip")]
+        public float SliderFloatTest { get; set; } = 0.1f;
+
+        [Slider(-1f, 10f, 1f, SliderAttribute.SliderType.Integer, description: "Test Int Slider Tooltip")]
+        public int SliderIntTest { get; set; } = 2;
+
+        [XmlIgnore]
+        [Button("Test Button Tooltip")]
+        public Action TestButton { get; set; } = () =>
```

Comment:
The action should receive the dialog, so it can operate on it. For example how would you implement an OK (Save) or Cancel button?

Reply:
You make the dialog in the function.
You do not need to make a Save / Cancel for the lib as that is handled already.
For edge cases / power users there are existing ways to get active dialogs.